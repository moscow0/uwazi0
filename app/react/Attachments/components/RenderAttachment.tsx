import React from 'react';
import { AttachmentSchema } from 'shared/types/commonTypes';
import ReactPlayer from 'react-player';
import { ClientFile } from 'app/istore';
import MarkdownMedia from 'app/Markdown/components/MarkdownMedia';
import { isSerializedFile, prepareHTMLMediaView } from 'shared/fileUploadUtils';

export const RenderAttachment = ({ attachment }: { attachment: AttachmentSchema | ClientFile }) => {
  const { mimetype = '' } = attachment;

  const fileURL = isSerializedFile(attachment) ? prepareHTMLMediaView(attachment) : attachment.url;

  if (mimetype.includes('image')) {
    return fileURL ? (
      <img src={fileURL} alt={attachment.originalname} />
    ) : (
      <img src={`/api/files/${attachment.filename}`} alt={attachment.originalname} />
    );
  }

  const isVideoAudio = mimetype.includes('video') || mimetype.includes('audio');

  const isFromSupportedSite = attachment.url && ReactPlayer.canPlay(attachment.url);

  if (isVideoAudio || isFromSupportedSite) {
    return fileURL ? (
      <MarkdownMedia config={`(${fileURL})`} />
    ) : (
      <MarkdownMedia config={`(/api/files/${attachment.filename})`} />
    );
  }

  return null;
};
