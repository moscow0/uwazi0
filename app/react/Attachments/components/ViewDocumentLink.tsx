import React, { Component } from 'react';
import { Link, withRouter, WithRouterProps } from 'react-router';
import { CurrentLocationLink } from 'app/Layout';
import { EntitySchema } from 'shared/types/entityType';

type Props = {
  filename: string;
  entity: EntitySchema;
};

type State = {};

export class ViewDocumentLinkBase extends Component<Props & WithRouterProps, State> {
  render() {
    const { filename, location, children, entity } = this.props;
    const onViewer = location.pathname.match(/entity/);
    return onViewer ? (
      <CurrentLocationLink
        className="btn btn-default"
        location={location}
        queryParams={{ file: filename, page: 1 }}
        type="button"
      >
        {children}
      </CurrentLocationLink>
    ) : (
      <Link
        className="btn btn-default"
        to={`/entity/${entity.sharedId}?file=${filename}`}
        type="button"
      >
        {children}
      </Link>
    );
  }
}

export const ViewDocumentLink = withRouter(ViewDocumentLinkBase);
