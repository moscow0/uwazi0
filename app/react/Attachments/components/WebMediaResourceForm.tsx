import { Field, LocalForm } from 'react-redux-form';
import { Translate } from 'app/I18N';
import { Icon } from 'UI';
import React from 'react';
import { FormGroup } from 'app/Forms';

interface WebMediaResourceFormProps {
  handleSubmit: (args: any) => void;
  dispatch?: (dispatch: Function) => void;
  url?: string | null;
  hasName?: boolean;
}

const WebMediaResourceForm = ({
  handleSubmit,
  url,
  dispatch,
  hasName = false,
}: WebMediaResourceFormProps) => {
  const validators = {
    ...(hasName && { name: { required: (val: any) => !!val && val.trim() !== '' } }),
    url: { required: (val: any) => !!val && val.trim() !== '' },
  };

  return (
    <LocalForm
      onSubmit={handleSubmit}
      getDispatch={dispatch}
      model="urlForm"
      validators={validators}
      initialState={{ url }}
      className={!hasName ? 'select-from-link' : ''}
    >
      <FormGroup className="has-feedback" model=".url">
        <Field model=".url">
          <input
            type="text"
            className="form-control web-attachment-url"
            placeholder="Paste URL here"
          />
        </Field>
      </FormGroup>
      {hasName && (
        <FormGroup className="form-group" model=".name">
          <Field model=".name" className="field">
            <input type="text" className="form-control web-attachment-name" placeholder="Title" />
          </Field>
        </FormGroup>
      )}
      <button type="submit" className="btn">
        <Icon icon="link" />
        &nbsp; <Translate>Add from URL</Translate>
      </button>
    </LocalForm>
  );
};

export { WebMediaResourceForm };
