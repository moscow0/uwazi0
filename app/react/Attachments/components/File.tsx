import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { Translate, t } from 'app/I18N';
import { language as transformLanguage, availableLanguages } from 'shared/languagesList';
import { isBlobFile } from 'shared/tsUtils';
import { Icon } from 'UI';
import { FileType } from 'shared/types/fileType';
import { APIURL } from 'app/config.js';
import { LocalForm, Control } from 'react-redux-form';
import { ClientBlobFile } from 'app/istore';
import { updateFile, deleteFile } from 'app/Attachments/actions/actions';
import { wrapDispatch } from 'app/Multireducer';
import { TocGeneratedLabel } from 'app/ToggledFeatures/tocGeneration';
import { NeedAuthorization } from 'app/Auth';
import { EntitySchema } from 'shared/types/entityType';
import { ViewDocumentLink } from './ViewDocumentLink';

type FileProps = {
  file: FileType | ClientBlobFile;
  storeKey: string;
  entity: EntitySchema;
  updateFile: (file: FileType, entity: Object) => any | void;
  deleteFile: (file: FileType, entity: Object) => any | void;
};
type FileState = {
  editing: boolean;
};

class File extends Component<FileProps, FileState> {
  static defaultProps = { updateFile: () => {}, deleteFile: () => {} };

  static contextTypes = {
    confirm: PropTypes.func,
  };

  constructor(props: FileProps) {
    super(props);
    this.state = {
      editing: false,
    };
    this.edit = this.edit.bind(this);
    this.cancel = this.cancel.bind(this);
    this.delete = this.delete.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  edit() {
    this.setState({ editing: true });
  }

  cancel() {
    this.setState({ editing: false });
  }

  delete() {
    this.context.confirm({
      accept: () => {
        this.props.deleteFile(this.props.file, this.props.entity);
        this.setState({ editing: false });
      },
      title: 'Confirm delete file',
      message: 'Are you sure you want to delete this file?',
    });
  }

  handleSubmit(file: FileType) {
    this.props.updateFile(file, this.props.entity);
    this.setState({ editing: false });
  }

  renderDeleteButton() {
    return (
      <button type="button" className="btn btn-outline-danger" value="Delete" onClick={this.delete}>
        <Icon icon="trash-alt" />
        &nbsp;
        <Translate>Delete</Translate>
      </button>
    );
  }

  renderFailed() {
    return (
      <div className="file-failed">
        <Icon icon="times" />
        &nbsp;
        <Translate>Conversion failed</Translate>
        <NeedAuthorization roles={['admin', 'editor']} orWriteAccessTo={[this.props.entity]}>
          {this.renderDeleteButton()}
        </NeedAuthorization>
      </div>
    );
  }

  renderReady() {
    const { language, filename = '' } = this.props.file;
    return (
      <div>
        <div>
          <span className="badge">
            <Translate>{language ? transformLanguage(language) || '' : ''}</Translate>
          </span>
          <TocGeneratedLabel file={this.props.file}>
            <Translate>ML TOC</Translate>
          </TocGeneratedLabel>
        </div>
        <div>
          <a
            href={`${APIURL}files/${filename}`}
            target="_blank"
            rel="noopener noreferrer"
            className="file-download btn btn-outline-secondary"
          >
            <Icon icon="cloud-download-alt" />
            &nbsp;
            <Translate>Download</Translate>
          </a>
          <NeedAuthorization roles={['admin', 'editor']} orWriteAccessTo={[this.props.entity]}>
            <button type="button" className="file-edit btn btn-outline-success" onClick={this.edit}>
              <Icon icon="pencil-alt" />
              &nbsp;
              <Translate>Edit</Translate>
            </button>
          </NeedAuthorization>
          <ViewDocumentLink filename={filename} entity={this.props.entity}>
            <Translate>View</Translate>
          </ViewDocumentLink>
        </div>
      </div>
    );
  }

  renderView() {
    const { originalname, status } = !isBlobFile(this.props.file)
      ? this.props.file
      : { originalname: (this.props.file as ClientBlobFile).originalFile.name, status: 'ready' };
    return (
      <div className="file">
        <div className="file-originalname">{originalname}</div>
        {status === 'ready' && this.renderReady()}
        {status === 'failed' && this.renderFailed()}
        {status === 'processing' && (
          <div>
            <Icon icon="clock" />
            &nbsp;
            <Translate>Processing</Translate>
          </div>
        )}
      </div>
    );
  }

  renderEditing() {
    const mapProps = {
      className: ({ fieldValue }: any) =>
        fieldValue.submitFailed && !fieldValue.valid ? 'form-control has-error' : 'form-control',
    };
    return (
      <LocalForm onSubmit={this.handleSubmit} initialState={this.props.file} className="file-form">
        <div className="form-group row">
          <div className="col-sm-12">
            <label htmlFor="originalname">
              <Translate>Document Title</Translate>&nbsp;*
            </label>
            <Control.text
              validators={{ required: (val: string) => Boolean(val && val.length) }}
              className="form-control"
              model=".originalname"
              id="originalname"
              mapProps={mapProps}
            />
          </div>
          <div className="col-sm-3">
            <label htmlFor="language">
              <Translate>Language</Translate>
            </label>
          </div>
          <div className="col-sm-9">
            <Control.select className="form-control" model=".language" id="language">
              {availableLanguages.map(language => (
                <option key={language.ISO639_3} value={language.ISO639_3}>
                  {language.localized_label} ({language.label})
                </option>
              ))}
              <option value="other">{t('System', 'other', 'other', false)}</option>
            </Control.select>
          </div>
          <div className="col-sm-4">
            <button type="submit" className="btn btn-outline-success">
              <Icon icon="save" />
              &nbsp;
              <Translate>Save</Translate>
            </button>
          </div>
          <div className="col-sm-4">
            <button
              type="button"
              className="btn btn-outline-secondary"
              value="Cancel"
              onClick={this.cancel}
            >
              <Icon icon="times" />
              &nbsp;
              <Translate>Cancel</Translate>
            </button>
          </div>
          <div className="col-sm-4">{this.renderDeleteButton()}</div>
        </div>
      </LocalForm>
    );
  }

  render() {
    return this.state.editing ? this.renderEditing() : this.renderView();
  }
}

const mapDispatchToProps = (dispatch: Dispatch<{}>, props: FileProps) =>
  bindActionCreators({ updateFile, deleteFile }, wrapDispatch(dispatch, props.storeKey));

export type { FileProps };
export { File };
export const ConnectedFile = connect(null, mapDispatchToProps)(File);
