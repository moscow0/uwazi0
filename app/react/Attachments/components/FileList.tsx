/* eslint-disable prefer-destructuring */
/* eslint-disable class-methods-use-this */
/* eslint-disable react/sort-comp */
import React, { Component } from 'react';

import { advancedSort } from 'app/utils/advancedSort';
import { Translate } from 'app/I18N';
import { FileType } from 'shared/types/fileType';
import { EntitySchema } from 'shared/types/entityType';
import UploadButton from 'app/Metadata/components/UploadButton';
import { NeedAuthorization } from 'app/Auth';
import languageLib from 'shared/languages';
import { ConnectedFile as File } from './File';
import './scss/filelist.scss';

const defaultProps = {
  files: [],
  entity: null,
  storeKey: '',
};

export type FileListProps = {
  files: Array<FileType>;
  entity: EntitySchema;
  storeKey: string;
};

export class FileList extends Component<FileListProps> {
  static arrangeFiles(files: Array<FileType> = []) {
    return advancedSort(files, { property: 'originalname' });
  }

  static defaultProps = defaultProps;

  renderFile(file: FileType, index: number) {
    const { storeKey, entity } = this.props;
    return (
      <li key={index}>
        <File file={file} storeKey={storeKey} entity={entity} />
      </li>
    );
  }

  orderFilesByLanguage(files: FileType[], systemLanguage: string) {
    const orderedFiles = [...files];
    const fileIndex = orderedFiles.findIndex(file => {
      const language = languageLib.get(file.language as string, 'ISO639_1');
      return language === systemLanguage;
    });
    if (fileIndex > -1) {
      const temp = orderedFiles[fileIndex];
      orderedFiles[fileIndex] = orderedFiles[0];
      orderedFiles[0] = temp;
    }
    return orderedFiles;
  }

  render() {
    const label = (
      <h2>
        <Translate>Primary Documents</Translate>
      </h2>
    );

    const { files, entity } = this.props;
    const orderedFiles = this.orderFilesByLanguage(files, entity.language as string);
    return (
      <div className="filelist">
        <div className="filelist-header">
          {orderedFiles.length === 0 ? (
            <NeedAuthorization roles={['admin', 'editor']} orWriteAccessTo={[entity]}>
              {label}
            </NeedAuthorization>
          ) : (
            <>{label}</>
          )}
          <div>
            <NeedAuthorization roles={['admin', 'editor']} orWriteAccessTo={[entity]}>
              <UploadButton
                entitySharedId={this.props.entity.sharedId}
                storeKey={this.props.storeKey}
              />
            </NeedAuthorization>
          </div>
        </div>
        <ul>{orderedFiles.map((file, index) => this.renderFile(file, index))}</ul>
      </div>
    );
  }
}

export default FileList;
