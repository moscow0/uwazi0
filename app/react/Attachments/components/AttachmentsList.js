import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Translate } from 'app/I18N';
import { advancedSort } from 'app/utils/advancedSort';
import { NeedAuthorization } from 'app/Auth';
import Attachment from 'app/Attachments/components/Attachment';

import UploadSupportingFile from './UploadSupportingFile';

export default class AttachmentsList extends Component {
  static arrangeFiles(files = []) {
    return advancedSort(files, { property: 'originalname' });
  }

  render() {
    const label = (
      <h2>
        <Translate>Supporting files</Translate>
      </h2>
    );

    const { parentId, parentSharedId, readOnly, storeKey, entity } = this.props;
    const forcedReadOnly = readOnly || Boolean(this.props.isTargetDoc);

    let uploadAttachmentButton = null;
    if (!this.props.isTargetDoc) {
      uploadAttachmentButton = (
        <NeedAuthorization roles={['admin', 'editor']} orWriteAccessTo={[entity]}>
          <div className="attachment-add">
            <UploadSupportingFile entitySharedId={this.props.parentSharedId} storeKey={storeKey} />
          </div>
        </NeedAuthorization>
      );
    }

    const attachments = AttachmentsList.arrangeFiles(this.props.attachments);
    return (
      <div className="attachments-list-parent">
        <div className="attachments-list-header">
          {attachments.length === 0 ? (
            <NeedAuthorization roles={['admin', 'editor']} orWriteAccessTo={[entity]}>
              {label}
            </NeedAuthorization>
          ) : (
            <>{label}</>
          )}
          {uploadAttachmentButton}
        </div>
        <div className="attachments-list">
          {attachments.map((file, index) => (
            <Attachment
              key={index}
              file={file}
              parentId={parentId}
              readOnly={forcedReadOnly}
              storeKey={storeKey}
              parentSharedId={parentSharedId}
              isSourceDocument={false}
              entity={entity}
            />
          ))}
        </div>
      </div>
    );
  }
}

AttachmentsList.defaultProps = {
  attachments: [],
  readOnly: false,
  isTargetDoc: false,
  parentId: null,
  parentSharedId: null,
  storeKey: '',
  entity: null,
};

AttachmentsList.propTypes = {
  attachments: PropTypes.arrayOf(PropTypes.object),
  parentId: PropTypes.string,
  parentSharedId: PropTypes.string,
  readOnly: PropTypes.bool,
  isTargetDoc: PropTypes.bool,
  storeKey: PropTypes.string,
  entity: PropTypes.object,
};

AttachmentsList.contextTypes = {
  confirm: PropTypes.func,
};
