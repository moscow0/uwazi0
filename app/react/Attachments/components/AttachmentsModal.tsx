import React, { useRef } from 'react';
import { connect } from 'react-redux';
import ReactModal from 'react-modal';
import { Tabs, TabLink, TabContent } from 'react-tabs-redux';
import Dropzone from 'react-dropzone';
import { bindActionCreators, Dispatch } from 'redux';
import { actions as formActions } from 'react-redux-form';
import { Translate } from 'app/I18N';
import { Icon } from 'app/UI';
import { WebMediaResourceForm } from 'app/Attachments/components/WebMediaResourceForm';

interface AttachmentsModalProps {
  isOpen: boolean;
  entitySharedId: string;
  storeKey: string;
  model: string;
  onClose(): void;
  uploadAttachment: (...args: any[]) => (dispatch: Dispatch<{}>) => Promise<any>;
  uploadAttachmentFromUrl: (...args: any[]) => (dispatch: Dispatch<{}>) => void;
  getPercentage?: number;
}

const AttachmentsModalCmp = ({
  isOpen,
  entitySharedId,
  storeKey,
  model,
  onClose,
  uploadAttachment: uploadAttachmentProp,
  uploadAttachmentFromUrl: uploadAttachmentFromUrlProp,
  getPercentage,
}: AttachmentsModalProps) => {
  const inputFileRef = useRef<HTMLInputElement | null>(null);
  let formDispatch: Function = () => {};

  const handleUploadButtonClicked = () => {
    if (inputFileRef.current) {
      inputFileRef.current.click();
    }
  };

  const handleInputFileChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
    if (event.target.files) {
      await Promise.all(
        [...event.target.files].map(file =>
          uploadAttachmentProp(entitySharedId, file, { __reducerKey: storeKey, model })
        )
      );
    }
  };

  const handleDropFiles = async (accepted: File[]) => {
    await Promise.all(
      accepted.map(file =>
        uploadAttachmentProp(entitySharedId, file, { __reducerKey: storeKey, model })
      )
    );
  };

  const handleSubmitUrlForm = (formModelData: { url: string; name: string }) => {
    uploadAttachmentFromUrlProp(entitySharedId, formModelData, { __reducerKey: storeKey, model });
    formDispatch(formActions.reset('urlForm'));
  };

  return (
    <ReactModal
      isOpen={isOpen}
      className="attachments-modal"
      overlayClassName="attachments-modal__overlay"
      ariaHideApp={false}
    >
      <div className="attachments-modal__header">
        <h4>
          <Translate>Supporting files</Translate>
        </h4>

        <button
          type="button"
          onClick={onClose}
          className="attachments-modal__close"
          disabled={getPercentage !== undefined}
        >
          <Icon icon="times" />
          <span>
            <Translate>Cancel</Translate>
          </span>
        </button>
      </div>
      <div className="attachments-modal__content">
        <Tabs renderActiveTabContentOnly>
          <div className="attachments-modal__tabs">
            <TabLink to="uploadComputer" className="tab-link modal-tab-1" component="div">
              <Translate>Upload from computer</Translate>
            </TabLink>
            <TabLink to="uploadWeb" className="tab-link modal-tab-2" component="div">
              <Translate>Add from web</Translate>
            </TabLink>
          </div>

          <div className="attachments-modal__tabs-content">
            <TabContent for="uploadComputer" className="tab-content centered">
              <Dropzone
                disableClick
                onDrop={handleDropFiles}
                className="attachments-modal__dropzone"
                multiple={false}
              >
                {getPercentage === undefined ? (
                  <>
                    <button type="button" onClick={handleUploadButtonClicked} className="btn">
                      <Icon icon="link" />
                      &nbsp; <Translate>Upload and select file</Translate>
                    </button>
                    <input
                      aria-label="fileInput"
                      type="file"
                      onChange={handleInputFileChange}
                      style={{ display: 'none' }}
                      ref={inputFileRef}
                    />
                    <h4 className="attachments-modal__dropzone-title">
                      <Translate>Drag and drop file in this window to upload </Translate>
                    </h4>
                  </>
                ) : (
                  <div className="progress attachments-modal-progress">
                    <div
                      className="progress-bar progress-bar-success attachments-modal-progress-bar"
                      role="progressbar"
                      style={{ width: `${getPercentage}%` }}
                    />
                  </div>
                )}
              </Dropzone>
            </TabContent>
            <TabContent for="uploadWeb" className="tab-content centered">
              <div className="wrapper-web">
                <WebMediaResourceForm
                  handleSubmit={handleSubmitUrlForm}
                  dispatch={(dispatch: Function) => {
                    formDispatch = dispatch;
                  }}
                  hasName
                />
              </div>
            </TabContent>
          </div>
        </Tabs>
      </div>
    </ReactModal>
  );
};

const mapDispatchToProps = (dispatch: Dispatch<{}>, ownProps: AttachmentsModalProps) =>
  bindActionCreators(
    {
      uploadAttachment: ownProps.uploadAttachment,
      uploadAttachmentFromUrl: ownProps.uploadAttachmentFromUrl,
    },
    dispatch
  );

const AttachmentsModal = connect(null, mapDispatchToProps)(AttachmentsModalCmp);

export type { AttachmentsModalProps };
export { AttachmentsModalCmp, AttachmentsModal };
