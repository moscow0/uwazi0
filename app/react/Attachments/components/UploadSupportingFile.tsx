import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { Translate } from 'app/I18N';
import { Icon } from 'UI';
import { attachmentCompleted } from 'app/Metadata/actions/supportingFilesActions';
import { uploadAttachment, uploadAttachmentFromUrl } from '../actions/actions';
import { AttachmentsModal } from './AttachmentsModal';

interface UploadSupportingFileProps {
  entitySharedId: string;
  storeKey: string;
  model?: string;
  progress?: any;
  uploadAttachment?: (...args: any[]) => (dispatch: Dispatch<{}>) => Promise<any>;
  uploadAttachmentFromUrl?: (...args: any[]) => (dispatch: Dispatch<{}>) => void;
  attachmentCompleted: (entity: string) => (dispatch: Dispatch<{}>) => void;
}

function mapStateToProps({ attachments }: { attachments: any }) {
  return {
    progress: attachments.progress,
  };
}

const mapDispatchToProps = (dispatch: Dispatch<{}>) =>
  bindActionCreators({ attachmentCompleted }, dispatch);

const connector = connect(mapStateToProps, mapDispatchToProps);

const UploadSupportingFile = (props: UploadSupportingFileProps) => {
  const { entitySharedId, storeKey, progress, model = '' } = props;
  const [modalOpen, setModalOpen] = useState(false);

  const openModal = () => setModalOpen(true);
  const closeModal = () => {
    setModalOpen(false);
    props.attachmentCompleted(entitySharedId);
  };

  const getPercentage = progress.get(entitySharedId);

  useEffect(() => {
    if (getPercentage === 100) {
      closeModal();
    }
  }, [progress]);

  return (
    <>
      <button type="button" onClick={openModal} className="btn attachments-modal-trigger">
        <Icon icon="plus" />
        &nbsp;
        <Translate>Add file</Translate>
      </button>

      <AttachmentsModal
        isOpen={modalOpen}
        onClose={closeModal}
        entitySharedId={entitySharedId}
        storeKey={storeKey}
        getPercentage={getPercentage}
        model={model}
        uploadAttachment={props.uploadAttachment || uploadAttachment}
        uploadAttachmentFromUrl={props.uploadAttachmentFromUrl || uploadAttachmentFromUrl}
      />
    </>
  );
};

export { mapDispatchToProps, mapStateToProps };
export default connector(UploadSupportingFile);
