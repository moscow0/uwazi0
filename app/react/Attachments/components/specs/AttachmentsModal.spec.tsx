import React from 'react';
import thunk from 'redux-thunk';
import { shallow, ShallowWrapper } from 'enzyme';
import ReactModal from 'react-modal';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { WebMediaResourceForm } from 'app/Attachments/components/WebMediaResourceForm';
import { AttachmentsModalCmp, AttachmentsModalProps } from '../AttachmentsModal';

const mockStore = configureMockStore([thunk]);
const store = mockStore({});

describe('Attachments Modal', () => {
  let component: ShallowWrapper;
  let props: AttachmentsModalProps;

  beforeEach(() => {
    props = {
      entitySharedId: '123',
      storeKey: '456',
      onClose: jasmine.createSpy('onClose'),
      uploadAttachmentFromUrl: jasmine.createSpy('uploadAttachmentFromUrl'),
      isOpen: true,
      uploadAttachment: jasmine.createSpy('uploadAttachment'),
      model: '',
    };
  });

  const render = (otherProps = {}) => {
    component = shallow(
      <Provider store={store}>
        <AttachmentsModalCmp {...props} {...otherProps} />
      </Provider>
    ).dive();
  };

  it('Should pass isOpen props to attachments modal.', () => {
    render({ isOpen: false });
    expect(component.find(ReactModal).props().isOpen).toBe(false);
    render({ isOpen: true });
    expect(component.find(ReactModal).props().isOpen).toBe(true);
  });

  it('Should match render of upload form', () => {
    render();

    expect(component).toMatchSnapshot();
  });

  it('Should match render of web form', () => {
    render();

    component.find('.modal-tab-2').simulate('click');

    expect(component).toMatchSnapshot();
  });

  it('Should submit web form', () => {
    render();

    component.find('.modal-tab-2').simulate('click');
    const form = component.find(WebMediaResourceForm).at(0);
    const formData = { url: 'http://test.test', name: 'testName' };
    form.props().handleSubmit(formData);
    expect(props.uploadAttachmentFromUrl).toHaveBeenCalledWith(props.entitySharedId, formData, {
      __reducerKey: props.storeKey,
      model: '',
    });
  });

  it('Should call onClose', () => {
    render();

    const closeButton = component.find('.attachments-modal__close');
    closeButton.simulate('click');

    expect(props.onClose).toHaveBeenCalled();
  });
});
