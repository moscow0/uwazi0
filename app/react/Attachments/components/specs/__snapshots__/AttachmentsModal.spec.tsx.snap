// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachments Modal Should match render of upload form 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="attachments-modal"
  closeTimeoutMS={0}
  contentElement={[Function]}
  isOpen={true}
  overlayClassName="attachments-modal__overlay"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="attachments-modal__header"
  >
    <h4>
      <Connect(Translate)>
        Supporting files
      </Connect(Translate)>
    </h4>
    <button
      className="attachments-modal__close"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <Connect(Icon)
        icon="times"
      />
      <span>
        <Connect(Translate)>
          Cancel
        </Connect(Translate)>
      </span>
    </button>
  </div>
  <div
    className="attachments-modal__content"
  >
    <Tabs
      onChange={[Function]}
      renderActiveTabContentOnly={true}
    >
      <div
        className="attachments-modal__tabs"
      >
        <TabLink
          className="tab-link modal-tab-1"
          component="div"
          to="uploadComputer"
        >
          <Connect(Translate)>
            Upload from computer
          </Connect(Translate)>
        </TabLink>
        <TabLink
          className="tab-link modal-tab-2"
          component="div"
          to="uploadWeb"
        >
          <Connect(Translate)>
            Add from web
          </Connect(Translate)>
        </TabLink>
      </div>
      <div
        className="attachments-modal__tabs-content"
      >
        <TabContent
          className="tab-content centered"
          for="uploadComputer"
        >
          <t
            className="attachments-modal__dropzone"
            disableClick={true}
            disablePreview={false}
            disabled={false}
            maxSize={Infinity}
            minSize={0}
            multiple={false}
            onDrop={[Function]}
            preventDropOnDocument={true}
          >
            <button
              className="btn"
              onClick={[Function]}
              type="button"
            >
              <Connect(Icon)
                icon="link"
              />
                
              <Connect(Translate)>
                Upload and select file
              </Connect(Translate)>
            </button>
            <input
              aria-label="fileInput"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
            <h4
              className="attachments-modal__dropzone-title"
            >
              <Connect(Translate)>
                Drag and drop file in this window to upload 
              </Connect(Translate)>
            </h4>
          </t>
        </TabContent>
        <TabContent
          className="tab-content centered"
          for="uploadWeb"
        >
          <div
            className="wrapper-web"
          >
            <WebMediaResourceForm
              dispatch={[Function]}
              handleSubmit={[Function]}
              hasName={true}
            />
          </div>
        </TabContent>
      </div>
    </Tabs>
  </div>
</Modal>
`;

exports[`Attachments Modal Should match render of web form 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="attachments-modal"
  closeTimeoutMS={0}
  contentElement={[Function]}
  isOpen={true}
  overlayClassName="attachments-modal__overlay"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="attachments-modal__header"
  >
    <h4>
      <Connect(Translate)>
        Supporting files
      </Connect(Translate)>
    </h4>
    <button
      className="attachments-modal__close"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <Connect(Icon)
        icon="times"
      />
      <span>
        <Connect(Translate)>
          Cancel
        </Connect(Translate)>
      </span>
    </button>
  </div>
  <div
    className="attachments-modal__content"
  >
    <Tabs
      onChange={[Function]}
      renderActiveTabContentOnly={true}
    >
      <div
        className="attachments-modal__tabs"
      >
        <TabLink
          className="tab-link modal-tab-1"
          component="div"
          to="uploadComputer"
        >
          <Connect(Translate)>
            Upload from computer
          </Connect(Translate)>
        </TabLink>
        <TabLink
          className="tab-link modal-tab-2"
          component="div"
          to="uploadWeb"
        >
          <Connect(Translate)>
            Add from web
          </Connect(Translate)>
        </TabLink>
      </div>
      <div
        className="attachments-modal__tabs-content"
      >
        <TabContent
          className="tab-content centered"
          for="uploadComputer"
        >
          <t
            className="attachments-modal__dropzone"
            disableClick={true}
            disablePreview={false}
            disabled={false}
            maxSize={Infinity}
            minSize={0}
            multiple={false}
            onDrop={[Function]}
            preventDropOnDocument={true}
          >
            <button
              className="btn"
              onClick={[Function]}
              type="button"
            >
              <Connect(Icon)
                icon="link"
              />
                
              <Connect(Translate)>
                Upload and select file
              </Connect(Translate)>
            </button>
            <input
              aria-label="fileInput"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
            <h4
              className="attachments-modal__dropzone-title"
            >
              <Connect(Translate)>
                Drag and drop file in this window to upload 
              </Connect(Translate)>
            </h4>
          </t>
        </TabContent>
        <TabContent
          className="tab-content centered"
          for="uploadWeb"
        >
          <div
            className="wrapper-web"
          >
            <WebMediaResourceForm
              dispatch={[Function]}
              handleSubmit={[Function]}
              hasName={true}
            />
          </div>
        </TabContent>
      </div>
    </Tabs>
  </div>
</Modal>
`;
